doc/input.md

```diff
@@ -275,6 +273,25 @@ class MyGame extends BaseGame with HasDraggableComponents {
 **Note**: `HasDraggableComponents` uses an advanced gesture detector under the hood and as explained
 further up on this page, shouldn't be used alongside basic detectors.
 
+### Hoverable components
+
+Just like the others, this mixin allows for easy wiring of your component to listen to hover states
+and events.
+
+By adding the `HasHoverableComponents` mixin to your base game, and by using the mixin `Hoverable` on
+your components, they get an `isHovered` field and a couple of methods (`onHoverStart`, `onHoverEnd`) that
+you can override if you want to listen to the events.
+
+```dart
+  bool isHovered = false; // note: you should not modify this value yourself
```

Comment:
Can't we transform this into a getter, so the user can't change it?

Reply:
it is a getter, they can always override it right?

Reply:
on the mixin level I can't control what the impl will do with it

Reply:
I thought it was a field, so da good then

Reply:
Actually, looking on the component, it is a field, I am confused

Reply:
not sure I understand your suggestions, we need a way to set it and a way to get it, it's a field on the mixin but can be overwritten by the user

Reply:
My suggestion is that this is a getter to the user of the component and internally, we set this using a setter that isn't exposed to the user. That way the user can't change the field at all.

Reply:
I thought more about it and there might be a way to hide it from the user! (slightly hide it)